src/main/client/scene/game-scene.ts

```diff
@@ -52,22 +60,40 @@ export class GameScene extends Scene {
          terrain.closePath();
          terrain.fillPath();
       }
-      terrain.generateTexture('terrain', sceneWidth, sceneHeight);
-      terrain.clear();
-      this.add.sprite(sceneWidth / 2, sceneHeight / 2, 'terrain');
+      this.terrainTexture = this.textures.createCanvas('terrain', sceneWidth, sceneHeight);
+      terrain.generateTexture(this.terrainTexture.getCanvas(), sceneWidth, sceneHeight);
+      this.add.sprite(0, 0, 'terrain');
    }
 
-   private hitTestTerrain(x: number, y: number, w: number, h: number): boolean {
-      for (let i = x; i < x + w; i++) {
-         for (let j = y; j < y + h; j++) {
-            if (this.textures.getPixelAlpha(i, j, 'terrain')) {
+   private hitTestTerrain(worldX: number, worldY: number, width: number, height: number): boolean {
+      const data = this.terrainTexture.getData(worldX, worldY, this.characterSize, this.characterSize2);
+      for (let i = 0; i < width; i++) {
+         for (let j = 0; j < height; j++) {
+            const pixel = this.getPixel(i, j, data);
+            if (pixel && pixel.green > 0) {
```

Comment:
We should use alpha if it does not influence performance.

Reply:
yup